test/JWT.hs

```diff
@@ -56,6 +66,69 @@ spec = do
     it "formats to a parsable and equal value" $
       decode (encode exampleClaimsSet) `shouldBe` Just exampleClaimsSet
 
+    describe "when the current time is prior to the Expiration Time" $
+      let
+        now = utcTime "2010-01-01 00:00:00"
+        run = flip runReader now
+      in
+        it "can be validated" $
+          run (validateClaimsSet conf exampleClaimsSet) `shouldBe` True
+
+    describe "when the current time is after the Expiration Time" $
+      let
+        now = utcTime "2012-01-01 00:00:00"
+        run = flip runReader now
+      in
+        it "cannot be validated" $
+          run (validateClaimsSet conf exampleClaimsSet) `shouldBe` False
+
+    describe "with a Not Before claim" $
+      let
+        claimsSet = emptyClaimsSet & claimNbf .~ intDate "2016-07-05 17:37:22"
+      in do
+        describe "when the current time is prior to the Not Before claim" $
+          let
+            now = utcTime "2015-01-01 00:00:00"
+            run = flip runReader now
+          in
+            it "cannot be validated" $
+              run (validateClaimsSet conf claimsSet) `shouldBe` False
+
+        describe "when the current time is after the Not Before claim" $
+          let
+            now = utcTime "2017-01-01 00:00:00"
+            run = flip runReader now
+          in
+            it "can be validated" $
+              run (validateClaimsSet conf claimsSet) `shouldBe` True
+
+    describe "with Expiration Time and Not Before claims" $
+      let
+        claimsSet = emptyClaimsSet & claimExp .~ intDate "2011-03-22 18:43:00"
+                                   & claimNbf .~ intDate "2011-03-20 17:37:22"
+      in do
+        describe "when the current time is prior to the Not Before claim" $
+          let
+            now = utcTime "2011-03-18 00:00:00"
+            run = flip runReader now
+          in
+            it "cannot be validated" $
+              run (validateClaimsSet conf claimsSet) `shouldBe` False
+        describe "when the current time is between the Not Before and Expiration Time claims" $
+          let
+            now = utcTime "2011-03-21 18:00:00"
+            run = flip runReader now
+          in
+            it "can be validated" $
+              run (validateClaimsSet conf claimsSet) `shouldBe` True
+        describe "when the current time is after the Expiration Time claim" $
+          let
+            now = utcTime "2011-03-24 00:00:00"
+            run = flip runReader now
+          in
+            it "cannot be validated" $
+              run (validateClaimsSet conf claimsSet) `shouldBe` False
+
```

Comment:
Should add tests for the "time is equal to nbf/exp" cases.  Current time equal to nbf should be accepted;
current time equal to exp should be rejected.